iceoryx_dust/container/include/iox/detail/fixed_position_container_helper.hpp

```diff
@@ -17,81 +17,146 @@
 #ifndef IOX_DUST_CONTAINER_DETAIL_FIXED_POSITION_CONTAINER_HELPER_HPP
 #define IOX_DUST_CONTAINER_DETAIL_FIXED_POSITION_CONTAINER_HELPER_HPP
 
+#include <functional>
 #include <utility>
 
 namespace iox
 {
 namespace detail
 {
-template <bool IsMove>
-struct AssignmentHelper;
 
-template <>
-struct AssignmentHelper<true>
+enum class SpecialCreationOperations
```

Comment:
I'm thinking about another name. I guess many developer won't immediately understand what this is for, including me if I would not have reviewed the last PR and potentially also my future self in a year again.

What do you think about replacing `SpecialCreation...` with `MoveAndCopy...`

Reply:
Good, it's more clear